src/sync-client.js

```diff
@@ -1166,159 +1187,149 @@ var self = {
+              //all pendings are resolved, the entry can be removed from meta data
+              delete meta[uid];
+            }
           }
         }
       }
-    }
-  },
+    },
 
 
-  markInFlightAsCrashed : function(dataset) {
-    var pending = dataset.pending;
-    var pendingHash;
-    var pendingRec;
+    markInFlightAsCrashed : function(dataset) {
+      var pending = dataset.pending;
+      var pendingHash;
+      var pendingRec;
 
-    if( pending ) {
-      var crashedRecords = {};
-      for( pendingHash in pending ) {
-        if( pending.hasOwnProperty(pendingHash) ) {
-          pendingRec = pending[pendingHash];
+      if( pending ) {
+        var crashedRecords = {};
+        for( pendingHash in pending ) {
+          if( pending.hasOwnProperty(pendingHash) ) {
+            pendingRec = pending[pendingHash];
 
-          if( pendingRec.inFlight ) {
-            self.consoleLog('Marking in flight pending record as crashed : ' + pendingHash);
-            pendingRec.crashed = true;
-            crashedRecords[pendingRec.uid] = pendingRec;
+            if( pendingRec.inFlight ) {
+              self.consoleLog('Marking in flight pending record as crashed : ' + pendingHash);
+              pendingRec.crashed = true;
+              crashedRecords[pendingRec.uid] = pendingRec;
+            }
           }
         }
       }
-    }
-  },
+    },
 
-  consoleLog: function(msg) {
-    if( self.config.do_console_log ) {
-      console.log(msg);
+    consoleLog: function(msg) {
+      if( self.config.do_console_log ) {
+        console.log(msg);
+      }
     }
-  }
-};
-
-(function() {
-  self.config = self.defaults;
-})();
-
-self.setCloudHandler(defaultCloudHandler.handler);
-
-module.exports = {
-  init: self.init,
-  manage: self.manage,
-  notify: self.notify,
-  doList: self.list,
-  getUID: self.getUID,
-  doCreate: self.create,
-  doRead: self.read,
-  doUpdate: self.update,
-  doDelete: self['delete'],
-  listCollisions: self.listCollisions,
-  removeCollision: self.removeCollision,
-  getPending : self.getPending,
-  clearPending : self.clearPending,
-  getDataset : self.getDataSet,
-  getQueryParams: self.getQueryParams,
-  setQueryParams: self.setQueryParams,
-  getMetaData: self.getMetaData,
-  setMetaData: self.setMetaData,
-  getConfig: self.getConfig,
-  setConfig: self.setConfig,
-  startSync: self.startSync,
-  stopSync: self.stopSync,
-  doSync: self.doSync,
-  forceSync: self.forceSync,
-  generateHash: self.generateHash,
-  loadDataSet: self.loadDataSet,
-  clearCache: self.clearCache,
-  doCloudCall: self.doCloudCall,
-  setCloudHandler: self.setCloudHandler,
-  setStorageAdapter: self.setStorageAdapter,
-  setHashMethod: self.setHashMethod,
-  setNetworkStatusHandler : self.setNetworkStatusHandler
-};
+  };
+
+  (function() {
+    self.config = self.defaults;
+  })();
+
+  self.setCloudHandler(defaultCloudHandler.handler);
+
+  return {
```

Comment:
This is the only other change. Obviously should return this object instead of exporting, since we're now exporting the factory function.